src/services/order.ts

```diff
@@ -1,86 +1,157 @@
-    baseURL: 'https://api.storelammoc.vn/orders',
-    timeout: 15e3,
-    headers: {'x-access-token': process.env.TOKEN_KEY},
+  baseURL: "https://api.storelammoc.vn/orders",
+  timeout: 15e3,
+  headers: { "x-access-token": process.env.TOKEN_KEY }
 });
 
-export async function getData(options: { since: Date, util: Date, limit?: number }) {
-    try {
-        const response = await request.get(`/customerAgencies/agencies/57e7af98b52ee4d36dc6c7c8`, {
-            params: {
-                apiVersion: 2,
-                ...options
-            }
-        })
-        return response.data
-    } catch (error) {
-        throw error
-    }
-
-}
-
-export async function processData() {
-    const a = await getData({since, limit: 10000, util})
-    let rawData = a.customerAgencyOrders;
-    let data = []
-
-    let variantIds = []
-    for (let z in rawData) {
-        for (let y in rawData[z].items) {
-            let item = rawData[z].items[y];
-            variantIds.push(item.variant.id)
+export async function fetchOrders(options: {
+  since: Date;
+  until: Date;
+  limit?: number;
+}): Promise<{ hasMore: boolean; customerAgencyOrders: any[] }> {
+  try {
+    const response = await request.get(
+      `/customerAgencies/agencies/57e7af98b52ee4d36dc6c7c8`,
+      {
+        params: {
+          apiVersion: 2,
+          ...options
         }
-    }
-
-    const variants = await getVariantByIds(variantIds);
-    const variantEntities = lodash.keyBy(variants, '_id')
-
-    for (let i in rawData) {
-        let order = rawData[i]
-        for (let x in rawData[i].items) {
-            let itemData = []
-            let item = rawData[i].items[x];
-            let productLink = 'https://storelammoc.vn/products/' + item.product.id + '?variantId=' + item.variant.id
-            let itemName = item.product.name + ' ( ' + item.variant.name + ' )'
-            let orderCreatedAt = moment(new Date(order.createdAt)).format('DD-MM-YYYY HH:mm')
-            itemData.push('CO' + order._id, orderCreatedAt, order.customerInfo.phone, order.customerInfo.name, lodash.get(variantEntities, `${item.variant.id}.sku`), itemName, item.quantity, item.price, order.status,)
-            data.push(itemData)
-        }
-    }
-
-    data.unshift(xlsxHeader)
-    // console.log(JSON.stringify(rawData[0], null, '\t'))
+      }
+    );
 
-    return data
+    return response.data;
+  } catch (error) {
+    throw error;
+  }
 }
 
-
-//check hasmore = true ? loop
-// check exist file in index ( when start up)
-//get orders, process orders => array data excell, exportxlsx
\ No newline at end of file
+export const generateXLSXOrderData = async (orders: any[]) => {
+  const variantIds = lodash.flatten(
+    orders.map(order => {
+      return order.items.map((item: any) => item.variant.id);
+    })
+  );
+
+  const variants = await getVariantByIds(variantIds);
+  const variantEntities = lodash.keyBy(variants, "_id");
+  let processedOrders = lodash.flatten(
```

Comment:
use `const`. `unshift` does not re-assign the `processedOrders` by a new value.

Reply:
Before version 2.8.0, the moment#subtract(String, Number) syntax was also supported. It has been deprecated in favor of moment#subtract(Number, String).